Sources/TuistKit/Generator/ModelLoader.swift

```diff
@@ -0,0 +1,27 @@
+import Foundation
+import Basic
+
+
+/// Entity responsible for provide generator models
+///
+/// Assumptions:
+///   - TuistGenerator creates a graph of Project dependencies
+///   - The Projects are associated with unique path
+///   - Each path only contains one Project
+///   - Whenever a dependency is encountered referencing another path,
+///     this entity is consulted again to load the model at that path
+protocol ModelLoading {
```

Comment:
This will move to `TuistGenerator` however as a first step introducing it to `TuistKit`